src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -163,13 +185,14 @@ private function evaluateBracket(string $expr, mixed $value): array
             return $result;
         }
 
-        // start, end and step
-        if (preg_match('/^(-?\d*):(-?\d*)(?::(-?\d+))?$/', $expr, $matches)) {
+        if (preg_match('/^(-?\d*+)\s*+:\s*+(-?\d*+)(?:\s*+:\s*+(-?\d++))?$/', $expr, $matches)) {
             if (!array_is_list($value)) {
                 return [];
             }
 
             $length = \count($value);
+            $matches = array_map('trim', $matches);
```

Comment:
not needed. The regex does not allow the matches to have leading or trailing spaces in matched groups, so this will be be a no-op.